CHAID/CHAID.py

```diff
@@ -396,16 +407,20 @@ def generate_best_split(self, ind, dep):
                         [col2_freq.get(k, 0) for k in keys]
                     ]
 
-                    chi = stats.chi2_contingency(np.array(cr_table), correction=False)
-                    sub_data[j] = (comb, chi[0], chi[1])
+                    ret = chi2(np.array(cr_table), correction=False)
+                    sub_data[j] = (comb, ret[1], ret[0])
 
-                choice, chi, highest_p_split = max(sub_data, key=lambda x: (x[2], x[1]))
+                choice, highest_p_join, _ = max(sub_data, key=lambda x: (x[1], x[2]))
 
-                if highest_p_split < self.alpha_merge:
+                if highest_p_join < self.alpha_merge:
+                    cr_table = [
```

Comment:
What's the linear as opposed to adding methods performance tradeoff we're making? Coz I'm now putting ANOTHER big chunk of code for the weighting. Grim. I think it will need to go in its own method as so much code duplication